firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -191,13 +229,20 @@ private void persistFid(String fid) throws FirebaseInstallationsException {
     }
   }
 
-  private Task<String> registerFidIfNecessary(PersistedFidEntry persistedFidEntry) {
+  /**
+   * Registers the FID with FIS servers if FID is in UNREGISTERED state.
+   *
+   * <p>Updates FID registration status to PENDING to avoid multiple network calls to FIS Servers.
+   */
+  private Task<String> registerFidIfNecessary(
+      PersistedFidEntry persistedFidEntry, AwaitListener listener) {
     String fid = persistedFidEntry.getFirebaseInstallationId();
 
     // Check if the fid is unregistered
     if (persistedFidEntry.getRegistrationStatus() == RegistrationStatus.UNREGISTERED) {
       updatePersistedFidWithPendingStatus(fid);
-      Tasks.call(executor, () -> registerAndSaveFid(persistedFidEntry));
+      Tasks.call(executor, () -> registerAndSaveFid(persistedFidEntry))
```

Comment:
There are additional check I can add . Ex: If Registered , listener onSuccess. I will do that once, I get this code reviewed. 

Thanks.